src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_b_fs_zyx_fsv16_imad.cpp

```diff
@@ -507,4 +532,23 @@ bool Convolution_kernel_b_fs_zyx_fsv16_imad::Validate(const Params& params) cons
 
     return true;
 }
+
+void Convolution_kernel_b_fs_zyx_fsv16_imad::GetUpdateDispatchDataFunc(KernelData& kd) const {
+    const auto& prim_params = static_cast<const convolution_params&>(*kd.params);
+    const auto& dynamicDispatchData = SetDefault(prim_params);
+
+    kd.update_dispatch_data_func = [this, dynamicDispatchData](const Params& params, KernelData& kd) {
```

Comment:
This approach looks better to me. However, I have one concern: will this work in case of model caching? It seems that in such a case kd.params might not be initialized, and we could capture incorrect parameters here?